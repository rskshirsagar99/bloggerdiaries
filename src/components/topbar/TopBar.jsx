
import { Link , useHistory } from "react-router-dom";
import "./topbar.css";
import firebaseConfig from "../../pages/initFirebase";
import React, { Component } from "react";

const user = firebaseConfig.auth().currentUser;
var username = sessionStorage.getItem("username")
export default class TopBar extends Component {
  constructor() {
    super();
    this.state = { isLoggedIn: false };
}

// Update navbar component when user logs in, to display "Login" or "Logout"
componentDidMount() {
    if (sessionStorage.getItem("isLoggedIn") === "true") {
        this.setState((prevState) => {
            if (!prevState.isLoggedIn) {
                return { isLoggedIn: true };
            }
        });
    }
    if (sessionStorage.getItem("isLoggedIn") === "false") {
        this.setState((prevState) => {
            if (prevState.isLoggedIn) {
                return { isLoggedIn: false };
            }
        });
    }
}
    
render() {
  return (
    <div className="top">
      <div className="topLeft">
      {this.state.isLoggedIn &&
            <span className="username">
            <span>{"Hi "+ username}</span>!
            </span>}
      </div>
      <div className="topCenter">
        <ul className="topList">
          <li className="topListItem">
            <Link className="link" to="/home">
              HOME
            </Link>
          </li>
          <li className="topListItem">
            <Link className="link" to="/about">
              ABOUT
            </Link>
          </li>
          <li className="topListItem">
            <Link className="link" to="/contact">
              CONTACT
            </Link>
          </li>
          <li className="topListItem">
            <a className="feedbackLink" href="https://forms.gle/ZHC31AN9dQLJ6X5NA" target="_blank">FEEDBACK</a>
          </li>
          {this.state.isLoggedIn && <li className="topListItem">
            <Link className="link" to="/write">
              WRITE
            </Link>
          </li>}

          {this.state.isLoggedIn && <li className="topListItem">
            <Link className="link" to="/stats">
              STATS
            </Link>
          </li>}
          
        </ul>
      </div>
      <div className="topRight">
          {!this.state.isLoggedIn ? (
            <span className="topListItem">
            <Link className="link" to="/login">
              LOGIN
            </Link>
          </span>
                ) : (
          <span className="topListItem">
              <Link
                  to="/logout"
                  className="link"
                  onClick={() => {
                      window.sessionStorage.removeItem(
                          "isLoggedIn"
                      );
                      window.sessionStorage.removeItem(
                          "username"
                      );
                      window.location = "/home";
                  }}
                      >
                          LOGOUT
                      </Link>
                  </span>
              )}
              {!this.state.isLoggedIn && <span className="topListItem">
              <Link className="link" to="/register">
                REGISTER
              </Link>
          </span>}
        
      </div>
    </div>
  );
  }
}
