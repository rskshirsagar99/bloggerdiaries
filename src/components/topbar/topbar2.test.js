import React from 'react'
import renderer from 'react-test-renderer'
import TopBar from './TopBar';

import { MemoryRouter } from 'react-router-dom';




test('snapshot test of topbar',()=>{

  const tree=renderer
  .create(
    <MemoryRouter>
      <TopBar/>
    </MemoryRouter>
  ).toJSON()
  expect(tree).toMatchSnapshot()


})