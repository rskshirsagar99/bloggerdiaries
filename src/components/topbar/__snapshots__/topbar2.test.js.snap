// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test of topbar 1`] = `
<div
  className="top"
>
  <div
    className="topLeft"
  >
    <ul
      className="topList"
    >
      <li
        className="topListItem"
      >
        <a
          className="link"
          href="/help"
          onClick={[Function]}
        />
      </li>
    </ul>
  </div>
  <div
    className="topCenter"
  >
    <ul
      className="topList"
    >
      <li
        className="topListItem"
      >
        <a
          className="link"
          href="/home"
          onClick={[Function]}
        >
          HOME
        </a>
      </li>
      <li
        className="topListItem"
      >
        <a
          className="link"
          href="/about"
          onClick={[Function]}
        >
          ABOUT
        </a>
      </li>
      <li
        className="topListItem"
      >
        <a
          className="link"
          href="/contact"
          onClick={[Function]}
        >
          CONTACT
        </a>
      </li>
      <li
        className="topListItem"
      >
        <a
          className="feedbackLink"
          href="https://forms.gle/ZHC31AN9dQLJ6X5NA"
          target="_blank"
        >
          FEEDBACK
        </a>
      </li>
      <li
        className="topListItem"
      >
        <a
          className="link"
          href="/write"
          onClick={[Function]}
        >
          WRITE
        </a>
      </li>
      <li
        className="topListItem"
      >
        <a
          className="link"
          href="/stats"
          onClick={[Function]}
        >
          STATS
        </a>
      </li>
      <li
        className="topListItem"
        onClick={[Function]}
      >
        LOGOUT
      </li>
    </ul>
  </div>
  <div
    className="topRight"
  >
    <a
      href="/settings"
      onClick={[Function]}
    />
  </div>
</div>
`;
